pkg/controllers/edgeconnect/controller.go

```diff
@@ -667,6 +666,32 @@ func (controller *Controller) createOrUpdateEdgeConnectDeployment(ctx context.Co
 	return nil
 }
 
-func edgeConnectClientSecretName(edgeConnectName string) string {
-	return edgeConnectName + "-client"
+func (controller *Controller) createOrUpdateEdgeConnectConfigSecret(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) (string, error) {
+	configFile, err := secret.PrepareConfigFile(ctx, edgeConnect, controller.apiReader)
+	if err != nil {
+		return "", err
+	}
+
+	secretData := make(map[string][]byte)
+	secretData[consts.EdgeConnectConfigFileName] = configFile
+
+	secretConfig, err := k8ssecret.Create(controller.scheme, edgeConnect,
+		k8ssecret.NewNameModifier(edgeConnect.Name+"-"+consts.EdgeConnectSecretSuffix),
+		k8ssecret.NewNamespaceModifier(edgeConnect.Namespace),
+		k8ssecret.NewDataModifier(secretData))
+
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	query := k8ssecret.NewQuery(ctx, controller.client, controller.apiReader, log)
+
+	err = query.CreateOrUpdate(*secretConfig)
+	if err != nil {
+		log.Info("could not create or update secret for edgeConnect.yaml", "name", secretConfig.Name)
+
+		return "", err
+	}
+
+	return hasher.GenerateHash(secretConfig.Data)
```

Comment:
Why not using the AddAnnotation() from hasher.go instead? This way you can generate the hash and add the annotation at the same time?

Reply:
because we want to add hash to `desiredDeployment.Spec.Template.Annotations`, but AddAnnotation works only with object itself.